packages/debug/src/browser/debug-session-manager.ts

```diff
@@ -210,7 +210,8 @@ export class DebugSessionManager {
     protected async startConfiguration(options: DebugConfigurationSessionOptions): Promise<DebugSession | undefined> {
         return this.progressService.withProgress('Start...', 'debug', async () => {
             try {
-                if (!await this.saveAll()) {
+                // If a parent session is available saving should be handled by the parent
+                if (!options.configuration.parentSessionId && !options.configuration.suppressSaveBeforeStart && !await this.saveAll()) {
```

Comment:
This seems to be a drive-by fix. Was our behaviour broken before? If so, what was the bug?

Reply:
No this is basically just what i described here:

> Note:
Parent/child sessions are currently handled as followed:

> For the saving supression only the root session is taken into account as this could have unintended side effects (saving files when not wanted), if a child session specifies something else than the parent session.

I wanted to make sure, that if the config that is executed specifies `suppressSaveBeforeStart: true` the save is never executed, e.g. if child configs do not specify the flag. WDYT?